execution_engine/src/core/runtime/auction_internal.rs

```diff
@@ -195,7 +195,12 @@ where
         .map_err(|_| Error::CLValue)?;
 
         let gas_counter = self.gas_counter();
-        let call_stack = self.call_stack().clone();
+        let mut call_stack = self.call_stack().clone();
```

Comment:
This is because `call_host_mint` assumes the call stack is already updated. We do push to call stack before creating a new Runtime object. I.e. in `Executor::exec` it already runs with an initial call stack (session + variant for stored contracts). In `Runtime::execute_contract` we're pushing a new element before `call_host_mint` is called. Auction internals are only outlier and without this change implemented checks wouldn't work consistently